examples/counter_app.rs

```diff
@@ -40,32 +40,39 @@ impl abci::Application for CounterApp {
         if c != self.count + 1 {
             resp.set_code(1);
             resp.set_log(String::from("Count must be incremental!"));
-            return resp;
+        } else {
+            // Update state to keep state correct for next check_tx call
+            self.count = c;
         }
 
-        // Update state to keep state correct for next check_tx call
-        self.count = c;
-        resp
+        let mut response = Response::new();
+        response.set_check_tx(resp);
+        let _ = responder.respond(response);
     }
 
-    fn deliver_tx(&mut self, req: &RequestDeliverTx) -> ResponseDeliverTx {
+    fn deliver_tx(&mut self, req: RequestDeliverTx, responder: Responder) {
         // Get the Tx [u8]
         let c = convert_tx(req.get_tx());
         // Update state
         self.count = c;
         // Return default code 0 == bueno
-        ResponseDeliverTx::new()
+        let res = ResponseDeliverTx::new();
+        let mut response = Response::new();
+        response.set_deliver_tx(res);
+        let _ = responder.respond(response);
```

Comment:
One other thought -- why throw away `Result<(), ()>` here and not return them?
if it's returned, https://github.com/tendermint/rust-abci/pull/95/files#diff-b4aea3e418ccdb71239b96952d9cddb6R329 could crash immediately (so there are no silent errors)

Reply:
You mean make it the return value of the trait method? In that case it would have to be a sync call, while now the app can pass the responder to other threads and eventually call `respond` on it.